### PR TITLE
Added upgrade functionality

### PR DESCRIPTION
**Pull request by koendelaat**

### CODE REVIEW

**Comment by koendelaat** [comment]
@nickhellings-philips Can you check this?

**Review by nickhellings — APPROVED** [review]

File: src/igtcloud/client/tools/cli.py [hunk 1]
```diff
@@ -170,9 +170,34 @@ def version():
...
+                package_version = redirect_url[ix + len(tag_prefix):]
+                if __version__ == package_version[1:]:
+                    print(f"Already at latest version: {package_version}")
+                    return
+    if package_version is None:
+        print("Cannot determine version")
+        return
+    import subprocess
+    import sys
+    subprocess.Popen([sys.executable, '-m', 'pip', 'install', '--upgrade', f'git+{__source__}@{package_version}'])
```
Comment (nickhellings) [inline src/igtcloud/client/tools/cli.py:194]: Doesn't work when using the executable.

Comment (koendelaat) [reply src/igtcloud/client/tools/cli.py]: @nickhellings Can you check again?

File: src/igtcloud/client/tools/cli.py [hunk 2]
```diff
@@ -170,9 +174,69 @@ def version():
...
+    if package_version is None:
+        print("Cannot determine version")
+        return
+
+    import sys
+    import subprocess
+    if getattr(sys, 'frozen', False) and hasattr(sys, '_MEIPASS'):
+        # running in a PyInstaller bundle
+        exe_path, exe_name = os.path.split(sys.executable)
+        url = f"{__source__}/releases/download/{package_version}/{exe_name}"
```
Comment (nickhellings) [inline src/igtcloud/client/tools/cli.py:202]: What happens if you rename the file though? You could determine based on the `if os.name == 'nt'` check you do below

File: src/igtcloud/client/tools/cli.py [hunk 3]
```diff
@@ -170,9 +174,69 @@ def version():
...
+                if __version__ == package_version[1:]:
+                    print(f"Already at latest version: {package_version}")
+                    return
+    if package_version is None:
+        print("Cannot determine version")
+        return
+
+    import sys
+    import subprocess
+    if getattr(sys, 'frozen', False) and hasattr(sys, '_MEIPASS'):
```
Comment (nickhellings) [inline src/igtcloud/client/tools/cli.py:199]: There's probably a reason so, what's the reason you do not do `if not sys.executable` or something like that ?

Comment (koendelaat) [reply src/igtcloud/client/tools/cli.py:199]: See: https://pyinstaller.org/en/stable/runtime-information.html

File: src/igtcloud/client/tools/cli.py (hunk 2 above)
Comment (koendelaat) [reply src/igtcloud/client/tools/cli.py:202]: Didn't want to put to much logic in this code.
If user renames the executable, than the upgrade functionality is broken.

**Pull request merged**